packages/truffle-codec-utils/src/compiler.ts

```diff
@@ -1,6 +1,19 @@
+import semver from "semver";
+
 export interface CompilerVersion {
   name?: string;
   version?: string;
   //NOTE: both these should really be present,
   //but they need to be optional for compilation reasons
 }
+
+export type SolidityFamily = "pre-0.5.0" | "0.5.x";
+
+export function solidityFamily(compiler: CompilerVersion): SolidityFamily {
+  if(semver.satisfies(compiler.version, "~0.5 || >=0.5.0", {includePrerelease: true})) {
```

Comment:
(Of course we'll have to change `>=` post v0.6.0)

Reply:
I think you can just use `^0.5.0`, no?

Reply:
The reason for the complex version condition is that I want to make sure to include prerelease versions of 0.5.0; I'm not sure that `>=0.5.0` or `^0.5.0` cover those, but I think `~0.5` does.  (Note the comment about how no, passing the `includePrerelease` flag is not sufficient for that.)

Thinking about it now, I guess it would make more sense to just use `~0.5`.  Only reason I didn't was because, well, this logic was factored from `isAddressPayable`, and it didn't occur to me to change it.  Could change it now or could save it for when we handle 0.6.0.

Reply:
Actually likely it makes sense to leave it as-is for now as if someone somehow *is* using Solidity 0.6.0 right now, it makes more sense to return 0.5.x rather than pre-0.5.0.  Yeah I think I'll just change this when we do 0.6.0 handling.